src/image-copy/azext_imagecopy/cli_utils.py

```diff
@@ -23,6 +23,11 @@ def run_cli_command(cmd, return_as_json=False):
 
         if return_as_json:
             if cmd_output:
+                # temp cleanup to deal with new warning string  - issue #979
+                # Related to: https://github.com/Azure/azure-cli/issues/10687
+                warn_string = "WARNING: The default kind for created storage account will change to 'StorageV2' from 'Storage' in future"
+                cmd_output = cmd_output.replace(warn_string, "")
+                # end of temp cleanup
```

Comment:
Although this is perfectly fine, IMO you can make the fix more robust by removing all chars before the first "{" char. What do you think?

Reply:
you would be correct, this is overly explicit to the problem. I'll adjust this and provide an update.

Thanks!

Reply:
I've modified it a bit to handle any text before the payload that is valid. However, we have to handle [ or { in that case, as the account keys for example returns an array.
Also added in debug logging for when it fires.

And as for that force-push, I just rebuilt my desktop and didn't set my email prior to commit. So I amended the author on the last commit.

Reply:
What do you think about something like this?

```phthon
        if return_as_json:
            if cmd_output:
                json_start_chars = {"{", "["}
                json_start_at = next((i for i, ch in enumerate(str(cmd_output)) if ch in json_start_chars), -1)
                logger.debug('json starts at position: %i ', json_start_at)
                if json_start_at > 0:
                    logger.debug("json output did not start at position 0, stripping the prefix.")
                    cmd_output = str(cmd_output)[json_start_at:]
                    logger.debug("after fix:")
                    logger.debug(cmd_output)
                json_output = json.loads(cmd_output)
                return json_output
```

Reply:
I like your approach there. I had thought about doing it off of position, wound up going the other way.

After chewing on mine a bit, there is a case it won't handle that yours will. I haven't seen that case returned by anything used by this extension yet. But, better to not assume it won't ever happen. :)

(Case in point, this code block can handle JSON objects, arrays of objects, and arrays of values where mine cannot handle arrays of values.)

I've pulled in the changes, with one minor change to the debug message to make them begin uniformly. Just in case anyone wants to grep that output for 'json output' simply.

Thanks for the help and great feedback!